### DISCUSSION TITLE
RLS with third-party auth solutions?

### DISCUSSION THREAD

**Discussion by akkie** [supabase/supabase#10012] [Questions] [answered]
Hi,

is it possible to use RLS with third-party auth solutions like Ory Kratos or is it limited only to Supabase's built-in auth solution?

Best regards,
Christian

**Comment by GaryAustin1** [comment, marked as answer]
In general you can use custom jwt's thru the API (with a few required fields like role, exp and sub).    https://supabase.com/docs/learn/auth-deep-dive/auth-deep-dive-jwts

If you don't have a jwt then you would likely need to do your own thing thru the standard Postgres data ports or have your own server API generate jwt's.  RLS is a Postgres concept independent of Supabase so could possibly work in some form with your auth info.

**Reply by akkie** [reply]
Thanks, will try issuing my own JWT with the required fields
